client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -203,6 +204,11 @@ class DeltaSharingRestClient(
     client
   }
 
+  private lazy val credentialProvider = AuthCredentialProviderFactory.createCredentialProvider(
```

Comment:
do we have to use a `CredentialProvider` class? what if we just define a `addAuthHeader` function from each profile class?

Reply:
That approach could work. However, I would argue that the role of a profile should primarily be to serve as a container for configuration rather than to implement the authentication logic. Moreover, if we need to introduce additional authentication-related methods in the future, the profile class could become bloated.